lua/vgit/core/Buffer.lua

```diff
@@ -326,4 +326,13 @@ function Buffer:set_keymap(mode, key, action)
   return self
 end
 
+function Buffer:store_buffer_vars()
+  local diff_status = self.git_object:generate_diff_status()
+  -- run inside vim.schedule or this fails with:
+  -- E5560: nvim_buf_set_var must not be called in a lua loop callback
+  vim.schedule(function()
+    vim.api.nvim_buf_set_var(self.bufnr, 'vgit_status', diff_status)
```

Comment:
I would take this line out and make a new method out of it and ditch the `store_buffer_vars` method:

```lua
function Buffer:set_var(name, value)
  vim.api.nvim_buf_set_var(self.bufnr, name, value)

  return self
end
```
The reason is to keep buffer methods as generic as possible and close to the `vim.api.nvim_buf*` functions and not couple it's logic around git logic (separation of concerns).

Reply:
done